Automattic/portkey-android
# Text Editor: Support text alignment

Closes #186. Adds a new text alignment button to the text editor dialog which cycles text alignment through left, center, and right-aligned:

![wp-stories-text-alignment](https://user-images.githubusercontent.com/9613966/88620387-dbf4fa80-d0d8-11ea-9732-c078378d79d2.gif)

I also made a few changes to the text editor layout, using a placeholder icon for the color picker (to be used properly with #465), and for now making it toggle the current color picker view so it doesn't overlap the alignment button.

I'm targeting the `feature/text-editing-v2` branch for now, which is branched off the alpha2 branch - once we've merged alpha 2 and shipped a new alpha this can be merged all the way to `develop`, but in the meantime I'll keep the text editing v2 work in the feature branch.

### Notes

I had to use the `gravity` property to align the EditText inside the editor, but `textAlignment` for the resulting TextView added to photoeditor (noted in a comment [here](https://github.com/Automattic/portkey-android/pull/466/files#diff-6e706ba2e92e5195fa534191694c2501R110)). The gravity handling is encapsulated, so outside the editor (e.g. for serialization) we're only talking about `textAlignment`. But still a bit confusing, I'm happy to try something else if you have any ideas @mzorz .

### To test

1. Add multi-line text to the editor, toggle the alignment with the button
2. Add the view to the canvas, make sure the alignment value is respected
3. Re-edit an existing text, change the alignment, save again, make sure all is persisted properly
4. Turn on Don't Keep Activities, verify that text alignment value is correctly stored and de-serialized
